src/dstack/_internal/cli/services/configurators/base.py

```diff
@@ -70,3 +75,29 @@ def register_args(cls, parser: argparse.ArgumentParser):
         args with different parser to show unified help.
         """
         pass
+
+
+class ApplyEnvVarsConfiguratorMixin(BaseApplyConfigurator):
```

Comment:
Any reason to inherit Mixin from BaseApplyConfigurator? I'd expect Mixin to not inherit from the base class.

Reply:
For type annotations only, no reason from the runtime point of view. I have no strong opinion on this, though.

Reply:
I see. I'd not depend on super to register mixin args but provide a method like register_env_args() similar to apply_env_vars() and call it from the child explicitly. It's easier to reason about and doesn't require to inherit the mixin.

Reply:
Agree, I'll update the PR.